Src/Plugins/Security/SEC101_020.DropboxAccessTokenValidator.cs

```diff
@@ -0,0 +1,133 @@
+                                 ref failureLevel,
+                                 ref fingerprint,
+                                 ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message, ref Dictionary<string, string> options)
+        {
+            return IsValidDynamic(Instance,
+                                  ref fingerprint,
+                                  ref message,
+                                  ref options);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            if (!groups.TryGetNonEmptyValue("refine", out string key))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            if (!ContainsDigitAndChar(key))
+            {
+                return nameof(ValidationState.NoMatch);
+            }
+
+            fingerprintText = new Fingerprint()
+            {
+                Key = key,
+                Platform = nameof(AssetPlatform.Dropbox),
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText, ref string message, ref Dictionary<string, string> options)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+            string key = fingerprint.Key;
+            using HttpClient httpClient = CreateHttpClient();
+            httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", key);
+
+            try
+            {
+                using HttpResponseMessage response = httpClient
+                    .PostAsync("https://api.dropboxapi.com/2/file_requests/count", null)
+                    .GetAwaiter()
+                    .GetResult();
+
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                    {
+                        return nameof(ValidationState.AuthorizedError);
+                    }
+
+                    case HttpStatusCode.BadRequest:
+                    {
+                        string body = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+
+                        // App deleted.
+                        if (body.Contains("This app is currently disabled"))
+                        {
+                            return nameof(ValidationState.Expired);
+                        }
+
+                        // Request was successfull but AccessToken does not have access.
+                        if (body.Contains("file_requests.read"))
```

Comment:
>Contains [](start = 33, length = 8)

Same here, this is a pretty obscure token 'file_requests.read', could be encode something more comprehensive? For really long strings that compromise readability, you can create const strings in the class.

Reply:
changed to const format and added a better message

---
In reply to: [598008335](https://github.com/microsoft/sarif-pattern-matcher/pull/315#discussion_r598008335) [](ancestors = 598008335)